staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go

```diff
@@ -55,9 +55,44 @@ var (
 	DefaultFeatureGate FeatureGate = NewFeatureGate()
 )
 
+// FeatureSpec describes a feature that can be used by many components
 type FeatureSpec struct {
-	Default    bool
+	// Default enables the feature by default
+	Default bool
+	// PreRelease set the release state of the feature
 	PreRelease prerelease
+	// Components list the components that is using the feature
+	Components []string
+}
+
+// Equals checks if two FeatureSpec are equal
+func (f FeatureSpec) Equals(other FeatureSpec) bool {
+	if f.Default != other.Default ||
+		f.PreRelease != other.PreRelease ||
+		len(f.Components) != len(other.Components) ||
+		(f.Components == nil) != (other.Components == nil) {
+		return false
+	}
+	sort.Strings(f.Components)
+	sort.Strings(other.Components)
+	for i, v := range f.Components {
+		if v != other.Components[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func (f FeatureSpec) hasComponent(component string) bool {
+	if len(f.Components) == 0 {
+		return true
+	}
+	for _, fc := range f.Components {
```

Comment:
what about:

```
_, ok := f.Components[component]
return ok
```

Reply:
Components is only a slice though, not a set so we cant use this. I think a slice is ok because its size is very small.

Reply:
Ah okay